soh/soh/OTRGlobals.cpp

```diff
@@ -1631,13 +1631,19 @@ extern "C" GetItemEntry ItemTable_RetrieveEntry(s16 tableID, s16 getItemID) {
 
 extern "C" GetItemEntry Randomizer_GetRandomizedItem(GetItemID ogId, s16 actorId, s16 actorParams, s16 sceneNum) {
     s16 getItemModIndex;
-    if (OTRGlobals::Instance->gRandomizer->CheckContainsVanillaItem(
-            OTRGlobals::Instance->gRandomizer->GetCheckFromActor(sceneNum, actorId, actorParams))) {
+    RandomizerCheck randomizerCheck = OTRGlobals::Instance->gRandomizer->GetCheckFromActor(sceneNum, actorId, actorParams);
+    if (OTRGlobals::Instance->gRandomizer->CheckContainsVanillaItem(randomizerCheck)) {
         getItemModIndex = MOD_NONE;
     } else {
         getItemModIndex = MOD_RANDOMIZER;
     }
     s16 itemID = OTRGlobals::Instance->gRandomizer->GetRandomizedItemId(ogId, actorId, actorParams, sceneNum);
+
+    // This feels awkward, maybe in the future each mod will have it's own obtainability check
```

Comment:
i feel like this might be less awkward if we were feeding it the RG instead of the RC

Reply:
As of #1379 I am able to do this if desired, but I don't  feel it alleviates my uncomfortable-ness here

Reply:
could you elaborate a bit on what makes you uncomfortable about it? we can make a cleanup issue for it and add it to the v3 milestone.

Reply:
I'm having trouble formulating it, but I think it's probably premature to consider how mods interact here so we can just ignore this for now